pkg/s3/s3.go

```diff
@@ -28,3 +57,185 @@ func GetConfig() *aws.Config {
 
 	return s3Config
 }
+
+// CreateS3BucketUsingAPod is helpful when trying to hit a cluster s3 service using the CLI since that could be used outside the cluster, or due to firewall restrictions
+func CreateS3BucketUsingAPod(ctx context.Context, clientset kubernetes.Interface, podOptions S3OpsPodOptions) error {
+	command := []string{"/s3-bucket-create.sh"}
+	pod, err := s3BucketPod(clientset, podOptions, command)
+	if err != nil {
+		return errors.Wrap(err, "failed to get pod resource")
+	}
+
+	createBucketPod, err := clientset.CoreV1().Pods(podOptions.Namespace).Create(ctx, pod, metav1.CreateOptions{})
+	if err != nil {
+		return errors.Wrap(err, "failed to create pod")
+	}
+
+	if err := k8sutil.WaitForPodCompleted(ctx, clientset, podOptions.Namespace, createBucketPod.Name, time.Minute*2); err != nil {
+		return errors.Wrap(err, "failed to wait for create bucket pod to complete")
+	}
+	defer clientset.CoreV1().Pods(podOptions.Namespace).Delete(ctx, createBucketPod.Name, metav1.DeleteOptions{})
+
+	logs, err := k8sutil.GetPodLogs(ctx, clientset, createBucketPod)
+	if err != nil {
+		return errors.Wrap(err, "failed to get pod logs")
+	}
+	if len(logs) == 0 {
+		return errors.New("no logs found")
+	}
+
+	type CreateBucketPodOutput struct {
+		Success bool `json:"success"`
+	}
+
+	createBucketPodOutput := CreateBucketPodOutput{}
+
+	scanner := bufio.NewScanner(bytes.NewReader(logs))
+	for scanner.Scan() {
```

Comment:
Why is this in a loop?  Maybe add a comment with example output? Also this could be a function with a unit test.

Reply:
what if there was any other output in there? like a warning or something? i wanted to just check for the line we're logging.